modules/videosipgw/VideoSIPGW.js

```diff
@@ -0,0 +1,144 @@
+        this.chatRoom = chatRoom;
+        this.eventEmitter = chatRoom.eventEmitter;
+        logger.info('creating VideoSIPGW');
+        this.sessions = {};
+
+        this.sessionStateChangeListener = this.sessionStateChanged.bind(this);
+
+        chatRoom.addPresenceListener('jibri-sip-status',
+            this.handleJibriSIPStatus.bind(this));
+        chatRoom.addPresenceListener('jibri-sip-call-state',
+            this.handleJibriSIPState.bind(this));
+    }
+
+    /**
+     * Handles presence nodes with name: jibri-sip-status.
+     *
+     * @param node the presence node to handle.
+     */
+    handleJibriSIPStatus(node) {
+        const attributes = node.attributes;
+
+        if (!attributes) {
+            return;
+        }
+
+        logger.log('Handle video sip gw status : ', attributes);
+        const newStatus = attributes.status;
+
+        // check for global availability of the service
+        if (newStatus !== this.status
+            && (newStatus === Constants.STATUS_UNDEFINED
+                || newStatus === Constants.STATUS_AVAILABLE
+                || newStatus === Constants.STATUS_BUSY)) {
+            this.status = newStatus;
+            this.eventEmitter.emit(
+                XMPPEvents.VIDEO_SIP_GW_AVAILABILITY_CHANGED, this.status);
+        }
+    }
+
+    /**
+     * Handles presence nodes with name: jibri-sip-call-state.
+     *
+     * @param node the presence node to handle.
+     */
+    handleJibriSIPState(node) {
+        const attributes = node.attributes;
+
+        if (!attributes) {
+            return;
+        }
+
+        logger.log('Handle video sip gw state : ', attributes);
+
+        const newState = attributes.state;
+
+        if (newState === this.state) {
+            return;
+        }
+
+        switch (newState) {
+        case Constants.STATE_ON:
+        case Constants.STATE_OFF:
+        case Constants.STATE_PENDING:
+        case Constants.STATE_RETRYING:
+        case Constants.STATE_FAILED: {
+            const address = attributes.sipaddress;
+
+            if (!address) {
+                return;
+            }
+
+            // find the corresponding session and set its state
+            const session = this.sessions[address];
+
+            if (session) {
+                session.setState(newState);
+            } else {
+                logger.warn('Video SIP GW session not found:', address);
+            }
+        }
+        }
+    }
+
+    /**
+     * Creates new session and stores its reference.
+     *
+     * @param {string} sipAddress - The sip address to use.
+     * @param {string} displayName - The display name to use.
+     * @returns {JitsiVideoSIPGWSession}
+     */
+    createVideoSIPGWSession(sipAddress, displayName) {
+        const session = new JitsiVideoSIPGWSession(
+            sipAddress, displayName, this.chatRoom);
+
+        session.addStateListener(this.sessionStateChangeListener);
+        this.sessions[sipAddress] = session;
```

Comment:
It would be good to check if we don't overwrite any session here and eventually print an error or something.